Site header (#595)
microsoft/atlas-design: site/src/components/site-header.md

---
title: Site header
description: The Site header component in the Atlas Design System
template: standard
classType: Component
classPrefixes:
  - site-header
---

# Site header

Site header component allows adding full width container to hold everything site's header might need - the logo, brand link, navigation, and other controls.

> **This page is best viewed with the content section full screened because banners are intended to be full width elements. Go full screen by clicking the button in the top right corner.**

Reviewer: I think this disclaimer isn't needed here. This doc looks good even on fairly narrow screens.

Author: Updated

## Usage

The site header is composed of the following elements:

| Component             | Description                                                 |
| --------------------- | ----------------------------------------------------------- |
| `site-header`         | The main container for the site header.                     |

> | `site-header-button`  | Buttons within the site header.                             |

Reviewer: Order here is a bit off, perhaps should be organized by DOM order and hierarchy?

1. site-header (parent)
1. site-header-logo
1. site-header-divider
1. site-header-nav
1. site-header-button

Author: Updated

| `site-header-logo`    | The logo displayed in the site header.                      |
| `site-header-divider` | A vertical divider between elements.                        |
| `site-header-brand`   | The brand name or title in the site header.                 |
| `site-header-nav`     | The navigation container within the site header.            |
| `site-header-panel`   | A panel that can be expanded to display additional content. |

> Below is the example of the site header with logo, brand and navigation links.

Reviewer: I am an unsufferable boor 😔, but I'm sorry I have to suggest the Oxford comma ...

```suggestion
Below is the example of the site header with logo, brand, and navigation links.
```

```html-no-indent
<div class="site-header">
	<!-- Microsoft logo -->

> 	<a href="https://www.microsoft.com" aria-label="Microsoft">

Reviewer: There's really no class here?

Author: Nope. `site-header-logo` applied directly to the image. Cause logo is not necessarily a link.

Reviewer: I think that's a miss on our end. It's quite odd to me to have a classless direct child of a component. The selector should probably be 
```
site-header-logo > svg

Author: Updated!

		<svg
			class="site-header-logo"
			aria-hidden="true"
			viewBox="0 0 25 25"
			fill="none"
			xmlns="http://www.w3.org/2000/svg"
		>
			<path d="M11.5216 0.5H0V11.9067H11.5216V0.5Z" fill="#f25022" />
			<path d="M24.2418 0.5H12.7202V11.9067H24.2418V0.5Z" fill="#7fba00" />
			<path d="M11.5216 13.0933H0V24.5H11.5216V13.0933Z" fill="#00a4ef" />
			<path d="M24.2418 13.0933H12.7202V24.5H24.2418V13.0933Z" fill="#ffb900" />
		</svg>
	</a>

	<!-- Divider -->
	<div class="site-header-divider"></div>

	<!-- Brand -->
	<a href="~/src/index.md" class="site-header-brand">
		<span>Brand</span>
	</a>

	<!-- Navigation -->
	<nav class="site-header-nav">
		<ul class="display-flex gap-xxs">
			<li>
				<a href="#!" class="site-header-button">
                    <span>Link 1</span>
                </a>
			</li>
			<li>
				<a href="#!" class="site-header-button">
                    <span>Link 2</span>
                </a>
			</li>
			<li>
				<a href="#!" class="site-header-button">
                    <span>Link 3</span>
                </a>
			</li>
		</ul>
	</nav>
</div>
```

### Logo

> There is an atomic available to position the logo right in the middle of header - `site-header-logo-centered`.

Reviewer: Since this class is paired with one element, it isn't strictly an atomic:
```suggestion
You can use ` site-header-logo-centered` subcomponent class to center the logo.
```

```html-no-indent
<div class="site-header justify-content-space-between">
	<!-- Brand -->
	<a href="~/src/index.md" class="site-header-brand">
		<span>Brand</span>
	</a>

    <!-- Microsoft logo -->
	<a href="https://www.microsoft.com" aria-label="Microsoft" class="site-header-logo-centered">
		<svg
			class="site-header-logo"
			aria-hidden="true"
			viewBox="0 0 25 25"
			fill="none"
			xmlns="http://www.w3.org/2000/svg"
		>
			<path d="M11.5216 0.5H0V11.9067H11.5216V0.5Z" fill="#f25022" />
			<path d="M24.2418 0.5H12.7202V11.9067H24.2418V0.5Z" fill="#7fba00" />
			<path d="M11.5216 13.0933H0V24.5H11.5216V13.0933Z" fill="#00a4ef" />
			<path d="M24.2418 13.0933H12.7202V24.5H24.2418V13.0933Z" fill="#ffb900" />
		</svg>
	</a>

	<a href="#!" class="link-button font-size-sm">
		<span>Link</span>
	</a>
</div>
```

### Panel

> There is a Panel sub-component available to use within site header. Panel provides additional space to hold more content. Panel is composed of the following elements:

Reviewer: de
```suggestion
There is a panel sub-component available to use within site header. Panel provides additional space to hold more content. Panel is composed of the following elements:
```

| Component                            | Description                                                                                          |
| ------------------------------------ | ---------------------------------------------------------------------------------------------------- |
| `site-header-panel`                  | The main container for the site header panel.                                                        |

> | `site-header-panel-featured-content` | Featured section in the panel. Located on the right side and it's recommended to hold up to 2 cards. |

Reviewer: ```suggestion
| `site-header-panel-featured-content` | Featured section in the panel. Located on the right side, this component holds up to two cards.|
```

> | `site-header-panel-content`          | Container that holds the main content. It should be either the set of links or cards.                |

Reviewer: ```suggestion
| `site-header-panel-content`          | Container that holds the main content. It should be either a set of links or cards.                |
```

| `site-header-panel-links`            | Section with set of links. Recommended to have up to 21 links.                                       |
| `site-header-panel-cards`            | Section with set of cards. Recommended to have up to 9 cards.                                        |

> Below are the examples of both content options - with a set of links and a set of cards:

Reviewer: Below are the examples of both content options, with a set of links and a set of cards:

> Below are the examples of both content options - with a set of links and a set of cards:

Reviewer: ```suggestion
Below are the examples of both content options, with a set of links and a set of cards:
```
